src/app/content/highlights/components/ConfirmationModal.tsx

```diff
@@ -1,47 +1,72 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
 import { FormattedMessage } from 'react-intl';
+import { useSelector } from 'react-redux';
 import styled from 'styled-components';
+import uuid from 'uuid';
 import Button from '../../../components/Button';
 import Modal from '../../../components/Modal';
 import { Body, BodyHeading, Footer } from '../../../components/Modal/styles';
-
-interface Props {
-  deny: () => void;
-  confirm: () => void;
-}
+import { assertDocument, assertNotNull } from '../../../utils';
+import { confirmationModalCallback, isConfirmationModalOpen } from '../selectors';
 
 // tslint:disable-next-line:variable-name
 const ConfirmationFooter = styled(Footer)`
   justify-content: space-between;
 `;
 
 // tslint:disable-next-line:variable-name
-const ConfirmationModal = ({deny, confirm}: Props) => {
-  return <Modal onModalClose={deny} heading='i18n:discard:heading'>
-    <Body>
-      <FormattedMessage id='i18n:discard:body'>
-        {(msg) => <BodyHeading>{msg}</BodyHeading>}
-      </FormattedMessage>
-    </Body>
-    <ConfirmationFooter>
-      <FormattedMessage id='i18n:discard:button:discard'>
-        {(msg) => <Button
-          data-testid='discard-changes'
-          onClick={confirm}
-          variant='primary'
+const ConfirmationModal = () => {
```

Comment:
I think it should be either named `DiscardChangesConfirmationModal` (actions, selectors, etc. should be the same) or it should be more generic by allowing to pass props that will handle displaying buttons and content

Reply:
I think that making it more generic is actually better, since you would have to just allow 4 props: heading, body, ok and cancel i18n keys. You can store them in a constant as an object so it will not be much work and we will be able to use this modal for other messages.